include/boost/unordered/detail/foa/core.hpp

```diff
@@ -1458,6 +1458,11 @@ table_core:empty_value<Hash,0>,empty_value<Pred,1>,empty_value<Allocator,2>
   using stats=table_core_stats;
 #endif
 
+#if defined(BOOST_GCC)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
```

Comment:
A `maybe-uninitialized` warning? What did the warning say, out of curiosity?

Historically, this might mean we need a launder() call somewhere.

Reply:
It is with this type of constructor:

```cpp
  table_core(
    std::size_t n=default_bucket_count,const Hash& h_=Hash(),
    const Pred& pred_=Pred(),const Allocator& al_=Allocator()):
    hash_base{empty_init,h_},pred_base{empty_init,pred_},
    allocator_base{empty_init,al_},arrays(new_arrays(n)),
    size_ctrl{initial_max_load(),0}
    {}
```
As it happens, `arrays` (in `arrays(new_arrays(n))`) is a non-static member function that uses the internal allocator, but _not_ other data members that have not been initialized yet (namely `size_ctrl`), but the compiler's analysis is not as fine-grained as that, seemingly. The situation would be similar to this example:

```cpp
struct X
{
  // calculate_b() called before X is fully initialized, but it's OK as b and c do not
  // form part of the calculation.
  
  X(int a_): a(a_), b(calculate_b()), c(0){}

  int calculate_b() { return 2*a; }
  
  int a;
  int b;
  int c;
};
```
What I find surprising is that this warning didn't trigger before, maybe it only happens for complex, non-inlined scenarios.